cmake/clang-format.cmake

```diff
@@ -0,0 +1,21 @@
+file(GLOB_RECURSE
+  ALL_CXX_SOURCE_FILES
+  *.[ch] *.[chi]pp *.[chi]xx *.cppm *.cc *.hh *.ii *.[CHI]
+)
+
+find_program(CLANG_FORMAT 
+  NAMES clang-format clang-format-9.0
+  clang-format-8.0 clang-format-7.0 clang-format-6.0)
+
+if(CLANG_FORMAT)
+  add_custom_target(
+    format
+    COMMAND ${CLANG_FORMAT}
+    -i
+    -style=file
+    ${ALL_CXX_SOURCE_FILES}
+    )
+  message(STATUS "Using ${CLANG_FORMAT} for target 'format'")
+else(CLANG_FORMAT)
+  message(WARNING "No suitable clang-format (>=6.0) found")
+endif(CLANG_FORMAT)
```

Comment:
Please don't repeat the condition in `else` and `endif` blocks, this is an obsolete CMake style.